Tutorials/Style/Functions-Docstrings.MD

```diff
@@ -0,0 +1,169 @@
+If you run this example in python, it should print the number 4, then the number 9. Here is how it works, line by line:
+1. I define the function with `def` and name the function `square`. Inside the parentheses, I give the function one input, `x`. Then, I put the colon `:` to tell python to expect an indented block with the function definition inside. 
+2. I use `return` to specify what the function should output when it is called, in this case, it returns `x**2`, the square of the input `x`
+3. I left this line blank for style reasons--to separate my function from the rest of the code
+4. I call the function `square` using the input `2`, using `square(2)` and wrap this in a `print()` call so I can see the output
+5. Similarly, I call the function `square` using the input `3`, using `square(3)` and wrap this in a `print()` call so I can see the output
+
+In general, a function is defined like this:
+
+```python
+def function_name(inputs, separated, by, commas):
+    code that turns your inputs 
+    into something you want to output
+    return output
+```
+
+### Understanding unfamiliar functions
+
+Often in scientific work, there are functions that other people have written and are available to us to use in our own code, but we don't know how to implement them (for instance,  mathematical functions in `numpy` and `scipy` modules or plotting functions in the `matplotlib` module). Fortunately, python has a built in way to get help on unfamiliar functions: the `help()` function!
+
+For instance, take the built in function `len`. If we don't know what this function does, we can simply put the name of the function into the `help()` function and get some useful information about it. Running
+
+```python
+help(len)
+```
+
+returns
+
+```
+Help on built-in function len in module builtins:
+
+len(obj, /)
+    Return the number of items in a container.
+```
+
+This tells us that `len` is a built in function that comes with python, and that it takes one input (an object) and returns the number of items in that container object. 
+
+`help()` also works on other objects, such as variables, which can be useful if you aren't sure what a particular variable is. For instance:
+
+```python
+x = [1, 2]
+help(x)
+```
+
+tells us that `x` is a list, and gives us all sorts of information about the sorts of methods and functions that can be used with lists. 
+
+___
+
+## Docstrings
+
+What happens if you write a function for your own code and then make it publicly available for other people to use? How will users know what your function does if they need to understand it or adapt it? Well, you could comment your code (always good practice), but for functions (and classes) in particular, you can use the magic of **_docstrings_**. 
+
+### How do I write a docstring?
+
+It's easy to write a docstring. Just under your function definition, add triple quotes ` ''' `, write your docstring, and then end the docstring with another set of triple quotes ` ''' `. For example: 
```

Comment:
Docstrings should be triple double quotes (`""" """`) according to this style guide: https://www.python.org/dev/peps/pep-0257/